### PR TITLE
feat: Add discord icon in header and footer

### LINKED ISSUE #4591: Add Discord Icon with Join Link to Header / Footer
As we've just shipped a discord server, it'd be nice to have it in the spaces that we have other external links for "social" sites.

Specifically, it'd be nice to have it in the header and / or the footer.

We'd probably want to have it after GitHub in the header if we want it there:
image

And we'd want to have it in the footer to the right of Twitter here:
image

### PR DESCRIPTION
**Pull request by parasg1999**
I have used SVG for discord logo to achieve the same animations as the github and twitter icons in header/footer. Since I had to add new class for managing the svg fill color, there may be changes required in the file where I have to place the appropriate classes.
Closes #4591

Suggestions are welcome. Posting screenshots as well (the animations are working, if you want I can post the GIF as well).

At header
Screenshot from 2020-10-05 04-25-46

At footer
Screenshot from 2020-10-05 04-26-11

### CODE REVIEW

**Review by erickzhao — CHANGES_REQUESTED** [review]
Seems like when I set up style linting, I didn't account for any files that weren't directly in `public/styles`. I'll have to set up a PR to fix that.

Also: seems like the icon is misaligned in the navbar on Chrome/FF/Safari on macOS image.

**Comment by parasg1999** [comment]
I don't know what's causing the issue and since I don't have access to MacOS, could you help me in fixing the bug?

**Comment by erickzhao** [comment]
Ref https://github.com/electron/electronjs.org/pull/4609 for aforementioned linter changes.

File: views/partials/footer.hbs [hunk 1]
```diff
@@ -7,6 +7,19 @@
...
+              title="The Discord Server">
+              <span class="vertical-middle">
+              <svg class="footer-discord" height="32px" enable-background="new 0 0 512 512" version="1.1"
+                viewBox="0 0 512 512" xml:space="preserve" xmlns="http://www.w3.org/2000/svg"
+                xmlns:xlink="http://www.w3.org/1999/xlink">
+                <circle cx="256" cy="256" fill="#20515b" id="ellipse" r="256" />
+                <path
+                  d="M372.4,168.7c0,0-33.3-26.1-72.7-29.1l-3.5,7.1c35.6,8.7,51.9,21.2,69,36.5  c-29.4-15-58.5-29.1-109.1-29.1s-79.7,14.1-109.1,29.1c17.1-15.3,36.5-29.2,69-36.5l-3.5-7.1c-41.3,3.9-72.7,29.1-72.7,29.1  s-37.2,54-43.6,160c37.5,43.3,94.5,43.6,94.5,43.6l11.9-15.9c-20.2-7-43.1-19.6-62.8-42.3c23.5,17.8,59.1,36.4,116.4,36.4  s92.8-18.5,116.4-36.4c-19.7,22.7-42.6,35.3-62.8,42.3l11.9,15.9c0,0,57-0.3,94.5-43.6C409.6,222.7,372.4,168.7,372.4,168.7z   M208.7,299.6c-14.1,0-25.5-13-25.5-29.1s11.4-29.1,25.5-29.1c14.1,0,25.5,13,25.5,29.1S222.8,299.6,208.7,299.6z M303.3,299.6  c-14.1,0-25.5-13-25.5-29.1s11.4-29.1,25.5-29.1s25.5,13,25.5,29.1S317.3,299.6,303.3,299.6z"
+                  fill="#f6f8f8" id="logo" /></svg>
```
Comment (vhashimotoo) [inline views/partials/footer.hbs:19]: Can you please move this code to own reusable file like `partials/icons/discord.hbs`, and reuse them by `{{> icons/discord}}`

File: views/partials/header.hbs [hunk 2]
```diff
@@ -33,6 +33,20 @@
...
+              title="The Discord Server">
+              <span class="vertical-middle">
+              <svg class="mega-octicon header-discord" height="32px" enable-background="new 0 0 512 512" version="1.1"
+                viewBox="0 0 512 512" xml:space="preserve" xmlns="http://www.w3.org/2000/svg"
+                xmlns:xlink="http://www.w3.org/1999/xlink">
+                <circle cx="256" cy="256" fill="#74b1be" id="ellipse" r="256" />
+                <path
+                  d="M372.4,168.7c0,0-33.3-26.1-72.7-29.1l-3.5,7.1c35.6,8.7,51.9,21.2,69,36.5  c-29.4-15-58.5-29.1-109.1-29.1s-79.7,14.1-109.1,29.1c17.1-15.3,36.5-29.2,69-36.5l-3.5-7.1c-41.3,3.9-72.7,29.1-72.7,29.1  s-37.2,54-43.6,160c37.5,43.3,94.5,43.6,94.5,43.6l11.9-15.9c-20.2-7-43.1-19.6-62.8-42.3c23.5,17.8,59.1,36.4,116.4,36.4  s92.8-18.5,116.4-36.4c-19.7,22.7-42.6,35.3-62.8,42.3l11.9,15.9c0,0,57-0.3,94.5-43.6C409.6,222.7,372.4,168.7,372.4,168.7z   M208.7,299.6c-14.1,0-25.5-13-25.5-29.1s11.4-29.1,25.5-29.1c14.1,0,25.5,13,25.5,29.1S222.8,299.6,208.7,299.6z M303.3,299.6  c-14.1,0-25.5-13-25.5-29.1s11.4-29.1,25.5-29.1s25.5,13,25.5,29.1S317.3,299.6,303.3,299.6z"
+                  fill="#2f3241" id="logo" /></svg>
```
Comment (vhashimotoo) [inline views/partials/header.hbs:46]: The same here

**Review by vhashimotoo — CHANGES_REQUESTED** [review]

**Comment by parasg1999** [comment]
@vhashimotoo Made the required changes, please check.

File: views/partials/footer.hbs [hunk 3]
```diff
@@ -7,6 +7,13 @@
                 class="mega-octicon octicon-mark-github vertical-middle"></span></a></li>
           <li><a class="footer-social-icon" href="https://twitter.com/electronjs" title="@electronjs on Twitter"><span
                 class="devicon-twitter-plain vertical-middle"></span></a></li>
+          <li><a class="footer-social-icon vertical-middle" href="https://discord.gg/H6uTh7m"
```
Comment (erickzhao) [inline views/partials/footer.hbs:10]:
```suggestion
          <li><a class="footer-social-icon vertical-middle" href="https://discord.gg/electron"
```

We got a vanity URL now :)

File: public/styles/helpers/_icons.scss [hunk 4]
```diff
@@ -68,3 +68,47 @@
...
+        fill: $jumbo-color-subtle;
+      }
+    }
+  }
+}
+
+.footer-discord {
+  svg {
+    circle {
+      color: $main-color-strong;
```
Comment (erickzhao) [inline public/styles/helpers/_icons.scss:101]:
```suggestion
      fill: $main-color-strong;
```

Renders as black using the `color` prop.

**Review by erickzhao — CHANGES_REQUESTED** [review]
Figured out the alignment issue, and a color issue in the footer. @parasg1999 could you test these fixes locally to see if they work for you as well?

**Comment by parasg1999** [comment]
@erickzhao The fix actually moved the discord icon above the github icon, but removing the octicon class worked for me, could you check if these changes work for MacOS?

**Review by erickzhao — CHANGES_REQUESTED** [review]
The fix indeed worked for me. :) 

screencast 2020-10-07 13-59-56

This is a direction in the right step, but the `.site-header-nav-item` class is leaving leftover hover styling.

https://github.com/electron/electronjs.org/blob/b0039fb951123cc1b2c24e81e8013f84d220185a/public/styles/helpers/_navigation.scss#L98-L99

I think we might want to clean this up by creating a separate `site-header-icon` class with similar styles to `.site-header-nav-item` minus the hover animation.

**Comment by parasg1999** [comment]
I added `site-header-nav-icon` class to the icon and used the scss rule for removing the hover.
Visually it still looks like the icon is above that of github but by using a [chrome extension](https://chrome.google.com/webstore/detail/baseliner/agoopbiflnjadjfbhimhlmcbgmdgldld/related?hl=en-GB) I was able to verify that they are aligned. I guess this is due to the circle not being complete in the github icon.

**Pull request closed without merging**